department-of-veterans-affairs/caseflow
# Education Regional Processing Office In Progress queue tab

Resolves [CASEFLOW-4603](https://vajira.max.gov/browse/CASEFLOW-4603)

### Description
Adds the In Progress tab for the Education Regional Processing Offices queues

### Acceptance Criteria
- [ ] Pre-docket appeals that have an `EducationAssessDocumentationTask` with a status of `in_progress` appear in the tab
- [ ] No other Pre-docket appeals should appear in the tab

### Testing Plan
1. Ensure that you have the `edu_predocket_appeals` feature toggle enabled which allows BVA Intake users to send Education appeals in to pre-docketing: `FeatureToggle.enable!(:edu_predocket_appeals)`
2. Intake a new 10182 appeal with a BVA Intake user (either `BVADWISE` or `BVAISHAW`)
3. When you reach the Add/Remove Issues screen, add an issue selecting "Education" as the benefit, selecting "Yes" for pre-docketing then submit the appeal
4. On the following success screen you should see a link to "edit issues" on the appeal, go ahead and navigate to that and grab the `uuid` for the appeal which you'll find in the current URL
5. At this point, we should have an appeal that has a `PreDocketTask` that is `on_hold` as well as an `EducationDocumentSearchTask` that is `assigned` to the `EducationEmo` organization
6. Since we lack the task action at this point that would normally allow us as an `EducationEmo` user to assign the appeal to one of our 3 `EducationRegionalProcessingOffice`s (Buffalo, Central Office, Muskogee) we instead need to manually create the task on our new appeal
7. Start a session of your rails console using `make c`
8. Look up the appeal with the `uuid` you grabbed in step 4 and save it inside a variable: `appeal = Appeal.find_by_uuid("<uuid>")`
9. Next we'll need to grab the `EducationDocumentSearchTask` as that will be the parent to the `EducationAssessDocumentationTask` that we're about to create.  You can accomplish this be examining the task tree on the appeal and locating the `ID` on the `EducationDocumentSearchTask` then saving it in a variable: `parent_task = Task.find(<id>)`
10.  Now let's grab one of our `EducationRegionalProcessingOffice`s so that we can assign them a task: `rpo = EducationRegionalProcessingOffice.first` (this should give us the Buffalo RPO)
11. At this point we should have everything we need to create our new task:
`EducationAssessDocumentationTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: rpo )`
12. Check your task tree and you should have an appeal with an `EducationAssessDocumentationTask` with a 
 status of `assigned` that is set as a child of the `EducationDocumentSearchTask`:
 <img width="955" alt="Screen Shot 2022-04-25 at 1 37 13 PM" src="https://user-images.githubusercontent.com/18618189/165143326-41a54496-2a09-4c50-92d9-ab44de45f49f.png">
13.  Go ahead and login as one our of Education Regional Processing Office users (either `EDURPOUSER` or `EDURPOADMIN`) and navigate to the Queue
14. From the "Switch views" dropdown, select the "Buffalo RPO team cases" option
15. Since we only have the "In Progress" tab at this point and the `EducationAssessDocumentationTask` has a status of `assigned` we should not see the appeal
16.  To successfully see our appeal, we need to update the status of our `EducationAssessDocumentationTask` to `in_progress`.  To do so, examine the task tree once again, grab the `ID` of the `EducationAssessDocumentationTask` and save it in another variable: `rpo_task = Task.find(<id>)`
17. Now we can update the status of the task by running: `rpo_task.update!(status: Constants.TASK_STATUSES.in_progress)`
18. Check your task tree again which should now look something similar to this:
 <img width="976" alt="Screen Shot 2022-04-25 at 1 47 28 PM" src="https://user-images.githubusercontent.com/18618189/165145017-4ab8c88d-e6ea-4e0a-89dc-a23537fc99de.png">
19.  At this point, the appeal should successfully show in the queue.  You can confirm it is indeed the correct appeal by navigating to the case details screen, grabbing the `uuid` of the appeal and comparing it to the `uuid` of the appeal you have in your rails console



### User Facing Changes
<img width="1236" alt="Screen Shot 2022-04-25 at 12 52 41 PM" src="https://user-images.githubusercontent.com/18618189/165139018-57497085-7e77-4a77-bc21-38ea6f4601c5.png">
